### DISCUSSION TITLE
Cannot delete Routing Table or Outbound Interconnection

### DISCUSSION THREAD

**Discussion by anthon-walters** [hnimminh/libresbc#143] [Q&A]
Hi,
Very new to LibreSBC. I have managed to get a basic installation running with the guides provided and its largely working as expected. I have now stumbled on an issue trying to delete a route that has an engagement with an Outbound Interconnect it doesn't want to release. I have tried to create a new route and tried to force the Outbound Interconnect there but  it doesn't update the change. When editing the Outbound Interconnect and trying to change the table in the engagement section to the new routing table again the update doesn't save and keeps on jumping back to the table I wish to delete. Any help will be greatly appreciated.

Outbound Interconnect section:
    ],
    "engagements": [
        "table:to_TSBC",
        "table:default_inbound_routing_to_TSBC"
    ]

The table "default_inbound_routing_to_TSBC" is the table I wish to delete and replace with table "to_TSBC" which already exists.

LibreSBC appears to circular reference these tables and engagements so that I cannot delete either the routing table or the Outbound Interconnection without getting the error "engaged routing  table" or "engaged outbound interconnection"

Here's wishing everyone a Happy New Year!

**Comment by anthon-walters** [comment]
Please ignore, seems I have mistakenly looked at the wrong Interconnection. Apologies for the noise! Love the project

**Reply by hnimminh** [reply]
No problem, good to hear that issue has been solved. 
Happy New Year!

**Reply by anthon-walters** [reply]
There is however one peculiarity that doesn't make sense. The following entry still exists in the "interconnection_to_IMPLYPBX

    "engagements": [
        "table:default_inbound_routing_to_pbx",
        "record:to_IMPLYITPBX:em:<obfuscated number>"
    ]

The table "default_inbound_routing_to_pbx" doesn't exist anymore. However when trying to remove the entry from the interconnection the data is accepted but it re-appears as soon as you enter the interconnection data screen again.

**Reply by hnimminh** [reply]
I will take a look, could you please describe:
* step how to reproduce the issue? 
* and what is version of libresbc are you using?

**Reply by anthon-walters** [reply]
I'm using version 0.7.1.a. The table was generated during the initial phase of deployment based on the examples probided by @khorsman . Once I started to understand the architecture and methodology I deleted the routing tables and their associated intercoonections and redid them with the naming convention I wanted to use. I battled a bit as per my first post but eventually understood where I was going wrong and managed to get the new naming structure going. Now when editing  "interconnection_to_IMPLYPBX" this old phantom table still appears under engagements. If I try to edit it out it accepts the edit with "Data has been submitted" but when I re-enter the Interconnection it is back under engagements.

Using WEB-UI with Go on my local machine (Windows) the LibreSBC is in a remote data centre.

**Comment by hnimminh** [comment]
This should be fixed by https://github.com/hnimminh/libresbc/issues/146
